spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitTemplateRetryTemplateConfigurer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.amqp;
+
+import org.springframework.boot.autoconfigure.retry.RetryTemplateConfigurer;
+
+/**
+ * A {@link RetryTemplateConfigurer} for RabbitMQ listeners.
+ *
+ * @author Gary Russell
+ * @since 2.0
+ *
+ */
+public class RabbitTemplateRetryTemplateConfigurer extends RabbitRetryTemplateConfigurer {
```

Comment:
This empty class seems suspicious.

Reply:
It's an empty class because there are two `RetryTemplate`s - one for the consumers and one for the template. 

How else can I determine that the user has provided a customizer for one Vs. the other?